src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import static edu.wpi.first.units.Units.Degrees;
+import static edu.wpi.first.units.Units.Radians;
+
+import com.ctre.phoenix6.configs.Slot0Configs;
+import com.ctre.phoenix6.configs.Slot1Configs;
+import com.ctre.phoenix6.controls.PositionDutyCycle;
+import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.ClimberConstants;
+import frc.robot.Constants.MotorIDConstants;
+import frc.robot.Robot;
+import java.util.function.BooleanSupplier;
+
+public class Climber extends SubsystemBase {
+  /** Creates a new Climber. */
+  private TalonFX climberMotorFront;
+
+  private TalonFX climberMotorBack;
+  private DCMotor simClimberGearbox;
+  private Mechanism2d simClimberArm;
+  private SingleJointedArmSim climberSim;
+  private Mechanism2d climbMech;
+  private MechanismRoot2d climbMechRoot;
+  private MechanismLigament2d climbMechLigament;
+  private Slot0Configs climberConfigsToClimber;
+  private Slot1Configs climberConfigsAtClimber;
+
+  public Climber() {
+    climberMotorFront = new TalonFX(MotorIDConstants.kCLimberMotorLeader);
+    climberMotorBack = new TalonFX(MotorIDConstants.kCLimberMotorFollower);
+    climberConfigsToClimber =
+        new Slot0Configs()
+            .withKP(ClimberConstants.kClimberP0)
+            .withKI(ClimberConstants.kClimberI0)
+            .withKD(ClimberConstants.kClimberD0)
+            .withKG(ClimberConstants.kClimberkG0)
+            .withKV(ClimberConstants.kClimberkV0);
+    climberConfigsAtClimber =
+        new Slot1Configs()
+            .withKP(ClimberConstants.kClimberP1)
+            .withKI(ClimberConstants.kClimberI1)
+            .withKD(ClimberConstants.kClimberD1)
+            .withKG(ClimberConstants.kClimberkG1)
+            .withKV(ClimberConstants.kClimberkV1);
+    climberMotorFront.getConfigurator().apply(climberConfigsToClimber);
+    climberMotorBack.getConfigurator().apply(climberConfigsToClimber);
+
+    if (Robot.isSimulation()) {
+      simClimberGearbox = DCMotor.getKrakenX60(1);
+      climberSim =
+          new SingleJointedArmSim(simClimberGearbox, 126, 0.077, 0.1, 0, Math.PI, false, 1);
```

Comment:
You'll want a second SingleJointedArm to use for when the cage is contacted and the climbing starts